http_check/datadog_checks/http_check/http_check.py

```diff
@@ -302,18 +329,28 @@ def check_cert_expiration(self, instance, timeout, instance_ca_certs):
             context.load_verify_locations(instance_ca_certs)
 
             ssl_sock = context.wrap_socket(sock, server_hostname=server_name)
-            cert = ssl_sock.getpeercert()
-            if cert:
-                exp_date = datetime.strptime(cert['notAfter'], "%b %d %H:%M:%S %Y %Z")
-            else:
+            binary_cert = ssl_sock.getpeercert(binary_form=True)
+
+            # To maintain backwards compatability, if we aren't validating tls/certs, do not process
+            # the returned binary cert unless specifically configured to with tls_retrieve_non_validated_cert
+            if (
+                instance.get("tls_verify") is False
```

Comment:
```suggestion
                not is_affirmative(instance.get("tls_verify", True))
```
Oops I forgot to leave this comment earlier

Reply:
updated